STUDIO/org.openl.rules.webstudio/webapp/pages/modules/topPanel.xhtml

```diff
@@ -161,62 +161,69 @@
-                    <li id="failuresSetting" class="#{!studio.testsFailuresOnly ? 'hidden' : ''}"
-                        onclick="event.stopPropagation()">
-                        <span tabindex="-1">
-                            <span>Failures per test</span>
-                            <h:selectOneMenu id="failures" value="#{studio.testsFailuresPerTest}">
-                                <f:selectItem itemLabel="1" itemValue="1" />
-                                <f:selectItem itemLabel="5" itemValue="5" />
-                                <f:selectItem itemLabel="20" itemValue="20" />
-                                <f:selectItem itemLabel="All" itemValue="-1" />
-                            </h:selectOneMenu>
-                        </span>
-                    </li>
-                    <li onclick="event.stopPropagation()">
-                        <span tabindex="-1" title="Display the full result of running test cases, not only values which are being tested">
-                            <span>Compound Result</span>
-                            <input name="complexResult" type="checkbox"
-                                value="true" checked="#{studio.showComplexResult ? 'checked' : ''}"
-                                style="vertical-align:-2px;margin:-4px" />
-                        </span>
-                    </li>
-                    <li class="divider"></li>
-                    <li>
-                        <span style="padding:6px 18px 7px">
-                            <a href="#{studio.url('test/test')}" class="button" onclick="test(event, '#{studio.url('test/test')}')">Test</a>
-                            <a href="#" id="downloadTest" class="button" onclick="downloadTest(event)">Test into File</a>
-                        </span>
-                    </li>
-                </ul>
-            </span>
-            <span class="delimeter"></span>
+            <h:panelGroup rendered="#{treeBean.canRun and !empty rProject and !empty studio.currentProjectDescriptor.modules}">
+                <h:panelGroup rendered="#{!empty module and treeBean.projectTestsCount > 0}">
+                    <h:outputLink value="#{studio.url('test/test')}" title="Run Tests"
+                                  styleClass="navbar-dropdown" onclick="test(event, '#{studio.url('test/test')}')">Test <span style="color:#888;padding: 1px 4px; border: 1px solid #ddd">#{treeBean.projectTestsCount}</span></h:outputLink>
+                </h:panelGroup>
+                <h:panelGroup rendered="#{empty module}">
+                    <c:set var="testUrl" value="#{utils.toUrl(rProject.repository.id, rProject.name, studio.currentProjectDescriptor.modules[0].name)}/test/test" />
+                    <h:outputLink value="#{testUrl}" title="Run Tests"
+                                  styleClass="navbar-dropdown" onclick="test(event, '#{testUrl}')">Test</h:outputLink>
+                </h:panelGroup>
+                <span class="dropdown dropdown-form">

```

Comment:
Do we still need this section for the case `#{!empty module and treeBean.projectTestsCount == 0}`?
Looks like we will render this panel for dropdown even if we we don't have "Run Tests" button (when there are no tests in the project)

Reply:
done